Drawer.d.ts

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react'
+
+export type DrawerChangeReason = 'swipe' | 'clickaway' | 'escape';
```

Comment:
Since it's only used in a callback, people might want to strong type it, something like:

```js

class Component extends React.Component<any, any> {

  // If we don't export the type people will have to manually declare the type
  // or it will be inferred as any. This is error-prone.
  private handleOnChange = (opening: boolean, reason: DrawerChangeReason) => {
    ...
  };

  render() {
    return <Drawer onRequestChange={this.handleOnChange}/>;
  }
}
```